utils/io.go

```diff
@@ -10,19 +10,22 @@ import (
 
 // WriteKeyToPemFile writes encode of key and purpose to the file
 func WriteKeyToPemFile(fileName string, keyPurpose string, key []byte) error {
-	f, err := os.Create(fileName)
+	// we are going with 0622 so that when umask is applied, we will get 0600 file permission
+	f, err := os.OpenFile(fileName, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0622)
```

Comment:
Does it work the same across OSs?

Reply:
There is no clear way of confirming this. The default Umask is 022 for most OS's. If someone change the umask to be more strict, then we will have problems with current(0644) or old (0666) as well. So I guess, it is okay

Reply:
wouldnt it be safer to create file first and then change the permissions? That way we overcome possible restrictive values of Umask?

Reply:
I think it should work